pkg/ccl/changefeedccl/helpers_test.go

```diff
@@ -676,6 +580,226 @@ func closeFeed(t testing.TB, f cdctest.TestFeed) {
 	}
 }
 
+// TestServer is a struct to allow tests to operate on a shared API regardless
+// of a test running as the system tenant or a secondary tenant, while still
+// having access to system interfaces for operations that require them such as
+// desctestutils.TestingGetPublicTableDescriptor
+type TestServer struct {
+	DB           *gosql.DB
+	Server       serverutils.TestTenantInterface
+	SystemDB     *gosql.DB
+	SystemServer serverutils.TestServerInterface
+	Codec        keys.SQLCodec
+	TestingKnobs base.TestingKnobs
+}
+
+func makeSystemServer(
+	t *testing.T, opts ...feedTestOption,
+) (testServer TestServer, cleanup func()) {
+	options := makeOptions(opts...)
+	return makeSystemServerWithOptions(t, options)
+}
+
+var _ = makeSystemServer // silence unused warning
+
+func makeSystemServerWithOptions(
+	t *testing.T, options feedTestOptions,
+) (testServer TestServer, cleanup func()) {
+	systemServer, systemDB, clusterCleanup := startTestFullServer(t, options)
+	return TestServer{
+			DB:           systemDB,
+			Server:       systemServer,
+			SystemDB:     systemDB,
+			SystemServer: systemServer,
```

Comment:
I'm not sure I like the fact that we make pretend set server and system server to the same values... I guess there are only  a few tests that want access to both tenant and host cluster.  I think I would have preferred not to export systemDB/Server.

I think it would have been better to have a method on TestServer... something like MustGetHostServer()... which could return those systemDB/server, but only if running in tenant mode.

Reply:
I'm not entirely sure what you mean.  For code that requires both Tenant and System servers such as `ProtectedTimestamp` code:
```go
ptp := s.Server.DistSQLServer().(*distsql.ServerImpl).ServerConfig.ProtectedTimestampProvider
store, err := s.SystemServer.GetStores().(*kvserver.Stores).GetStore(s.SystemServer.GetFirstStoreID())
require.NoError(t, err)
ptsReader := store.GetStoreConfig().ProtectedTimestampReader
```
We need to individually access a server that could either be tenant or system, then we need to access a server that's definitely the system server and is typed accordingly.  If we had a method that only returned the system db / server while running in tenant mode then the test itself would need conditional logic to refer to `MustGetHostServer()`while in tenant mode but `Server` in system mode, and we wouldn't be able to have `Server` always return a `TestTenantInterface`.

By setting server and system server to the same values, I wouldn't say we're really "pretending" since `server` and `DB` mean "the server and db of the current tenant", and that tenant happens to be the System tenant (and is still seen as just a TestTenantInterface).

Reply:
Perhaps pretending was not the right word.  I'm okay with using same value.  I'm saying that by default, we shouldn't expose systemDB/systemServer.  The tests shouldn't care -- as you say above.
Only in the (hopefully few?) cases when you have a test that needs access to both the tenant and the host (i.e. those tests know for a fact that they are tenant only tests), only then you'd want to have access to the underlying
system db.  And that's where I suggest we should be more explicit and have MustGetHostServer (or some such).

Reply:
> What if we had `TestServer` not include `SystemServer` and `SystemDB` and instead have a

That would be fine.  I just want to make sure things are explicit.